### PR TITLE
mapserver: new package

### PR DESCRIPTION
**Pull request by neilflood**
I confess I have only tested the client-side part of this, and only the Python binding. The rest builds fine, but I have not actually tried to run a server with the result.

### CODE REVIEW

File: var/spack/repos/builtin/packages/mapserver/package.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    depends_on('python', when='+python')
+    depends_on('ruby', when='+ruby')
+    depends_on('java', when='+java')
+    depends_on('perl', when='+perl')
+
+    variant('python', default=False)
+    variant('curl', default=False)
+    variant('ruby', default=False)
+    variant('java', default=False)
+    variant('perl', default=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mapserver/package.py]: Move the variants above the dependencies. Also, each variant needs a description.

File: var/spack/repos/builtin/packages/mapserver/package.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            args.append('-DWITH_RUBY=OFF')
+
+        if '+perl' in self.spec:
+            args.append('-DWITH_PERL=ON')
+        else:
+            args.append('-DWITH_PERL=OFF')
+
+        # These things are switched on by default, although possibly some
+        # should be variants.
+        args.append('-DWITH_CURL=ON')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mapserver/package.py]: You do have a variant for this one.

Comment (neilflood) [reply var/spack/repos/builtin/packages/mapserver/package.py]: Oops :-) Thank you for catching my omissions

File: var/spack/repos/builtin/packages/mapserver/package.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    homepage = "http://www.mapserver.org/"
+    url      = "http://download.osgeo.org/mapserver/mapserver-7.2.1.tar.gz"
+
+    version('7.2.1', sha256='9459a7057d5a85be66a41096a5d804f74665381186c37077c94b56e784db6102')
+
+    variant('python', default=False)
+    variant('curl', default=False)
+    variant('ruby', default=False)
+    variant('java', default=False)
+    variant('perl', default=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mapserver/package.py]: These variants need descriptions. @alalazo should we make `description` a required arg? I think so.

**Comment by adamjstewart** [comment]
```
var/spack/repos/builtin/packages/mapserver/package.py:24: [E501] line too long (97 > 79 characters)
var/spack/repos/builtin/packages/mapserver/package.py:57: [W605] invalid escape sequence '\$'
```

**Comment by neilflood** [comment]
re: the flake8 messages. Yes, I was aware, but unsure of the best way to handle them, particularly the escape sequence message. Could you recommend the right approach? Happy to do whatever you suggest.

**Comment by adamjstewart** [comment]
For escape sequence, try converting it to a raw string: `r'\$...'`

**Comment by adamjstewart** [comment]
For long line, if you keep the description on the same line, Spack will automatically ignore it.

**Pull request merged**